sway-core/src/lib.rs

```diff
@@ -606,13 +616,14 @@ pub fn compile_to_ast(
 
         // Check if we can re-use the data in the cache.
         if is_parse_module_cache_up_to_date(engines, &path, include_tests) {
-            let mut entry = query_engine.get_programs_cache_entry(&path).unwrap();
+            let mut entry = query_engine
+                .get_programs_cache_entry(&path)
+                .unwrap_or_else(|| panic!("unable to find entry in cache at path {:?}", &path));
```

Comment:
I was originally experimenting with not cloning the QE between compilation events as it takes the longest time and I feel like we should probably not be needing to...? I was getting panics at this location during this time though hence the reason I added this. But we are now cloning the QE engine as well so i'll remove this and put it back to just use an `unwrap`. It would be interesting to see if we could find a way to use an `Arc` around the types in the QE though. LIke:

```rust
pub struct QueryEngine {
    parse_module_cache: Arc<RwLock<ModuleCacheMap>>,
    programs_cache: Arc<RwLock<ProgramsCacheMap>>,
}
```